### PR TITLE
Bind variables even tho there is no variables in the original workflow xml

### PR DESCRIPTION
**Pull request by paraita**
See #3203 for the details

### CODE REVIEW

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java [hunk 1]
```diff
@@ -346,6 +347,16 @@ public JobType getType() {
         try {
             int eventType;
             boolean shouldContinue = true;
+
+            if (replacementVariables != null) {
+                commonPropertiesHolder.getVariables()
+                                      .putAll(replacementVariables.entrySet()
+                                                                  .stream()
+                                                                  .collect(Collectors.toMap(entry -> entry.getKey(),
```
[the remark this answers is not in the document]
Comment (jrochas) [reply scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java]: +1

**Comment by paraita** [comment]
retest this please

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java [hunk 2]
```diff
@@ -41,6 +41,9 @@
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java]: MINOR Remove this unused import 'java.util.function.Function'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java [hunk 3]
```diff
@@ -41,6 +41,9 @@
...
+import java.util.stream.Collector;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/StaxJobFactory.java]: MINOR Remove this unused import 'java.util.stream.Collector'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**